nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -135,17 +143,51 @@ LifecycleManager::destroyLifecycleServiceClients()
   }
 }
 
+bool
+LifecycleManager::createBondConnection(const std::string & node_name)
+{
+  const double timeout_ns =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(bond_timeout_).count();
+  const double timeout_s = timeout_ns / 1e9;
+
+  if (bond_map_.find(node_name) == bond_map_.end() && bond_timeout_.count() > 0.0) {
+    bond_map_[node_name] =
+      std::make_shared<bond::Bond>("bond", node_name, bond_client_node_);
+    bond_map_[node_name]->setHeartbeatTimeout(timeout_s);
+    bond_map_[node_name]->setHeartbeatPeriod(0.10);
+    bond_map_[node_name]->start();
+    if (!bond_map_[node_name]->waitUntilFormed(rclcpp::Duration(timeout_ns / 2))) {
```

Comment:
This is where the lifecycle managers throw `terminate called after throwing an instance of 'statemap::StateUndefinedException'`

Reply:
Yeah see the bond PR, I just forgot to push a commit that adds a sleep